src/Compilers/VisualBasic/Portable/Syntax/InternalSyntax/SyntaxTrivia.vb

```diff
@@ -49,6 +49,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             ObjectBinder.RegisterTypeReader(GetType(SyntaxTrivia), Function(r) New SyntaxTrivia(r))
         End Sub
 
+        Friend Overrides ReadOnly Property ShouldReuseInSerialization As Boolean
+            Get
+                Select Case Me.Kind
+                    Case SyntaxKind.WhitespaceTrivia,
+                        SyntaxKind.EndOfLineTrivia,
+                        SyntaxKind.LineContinuationTrivia,
+                        SyntaxKind.DocumentationCommentExteriorTrivia,
+                        SyntaxKind.ColonTrivia
+
+                        Return True
```

Comment:
>True [](start = 31, length = 4)

Should we delegate to base here? Also, does the size of the trivia matter?

Reply:
The base filers out nodes with diagnostics, doc comments and such. That generally does not apply to trivia. We can call that, but it will not do a lot of difference.

It does not look like size matters for when VB caches the trivia. The only kind with nontrivial size is Whitespace and it looks like it is cached regardless of size.

---
In reply to: [181835083](https://github.com/dotnet/roslyn/pull/26156#discussion_r181835083) [](ancestors = 181835083)